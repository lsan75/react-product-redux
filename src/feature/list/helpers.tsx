import { IProduct } from '../../store/product/product'

export const productList: IProduct[] = [
  {
    name: 'aaa',
    price: 10,
    selected: true
  },
  {
    name: 'bbb',
    price: 20,
    selected: false
  },
  {
    name: 'ccc',
    price: 50
  }
]