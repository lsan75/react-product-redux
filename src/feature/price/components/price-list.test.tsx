import * as Adapter from 'enzyme-adapter-react-16'
import * as React from 'react'

import { configure, shallow, ShallowWrapper } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import PriceList from './price-list'

import { productList } from '../../../mocks/helpers'

describe('PriceHeader', () => {
  configure({ adapter: new Adapter() })
  let shallowOutput: ShallowWrapper

  beforeEach(() => {
    shallowOutput = shallow(
      <PriceList products={productList} />
    )
  })

  afterEach(() => {
    shallowOutput.unmount()
  })

  it('should render', () => {
    expect(shallowToJson(shallowOutput)).toMatchSnapshot()
  })
})