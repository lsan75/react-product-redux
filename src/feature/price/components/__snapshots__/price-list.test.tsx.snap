// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceHeader should render 1`] = `
<ul
  className="PriceList"
>
  <li
    className="PriceList__Item"
    key="aaa"
  >
    <strong
      className="PriceList__Item__Name"
    >
      aaa
    </strong>
    <span
      className="PriceList__Item__Price"
    >
      10
       €
    </span>
  </li>
  <li
    className="PriceList__Item"
    key="bbb"
  >
    <strong
      className="PriceList__Item__Name"
    >
      bbb
    </strong>
    <span
      className="PriceList__Item__Price"
    >
      20
       €
    </span>
  </li>
  <li
    className="PriceList__Item"
    key="ccc"
  >
    <strong
      className="PriceList__Item__Name"
    >
      ccc
    </strong>
    <span
      className="PriceList__Item__Price"
    >
      50
       €
    </span>
  </li>
</ul>
`;
